src/general.rs

```diff
@@ -249,16 +250,23 @@ pub fn initialize(
                     token_types: ctx
                         .config
                         .semantic_tokens
-                        .keys()
+                        .iter()
                         .cloned()
-                        .map(|x| x.into())
+                        .map(|token| token.name.into())
+                        // Collect into HashSet first to remove duplicates
```

Comment:
I'd probably say "a set" instead of "HashSet" in the comment because it's an implementation detail.
For extremely large data, a BTreeSet would probably be more efficient but that's only a guess and it definitely doesn't matter here.